skada/_reweight.py

```diff
@@ -669,3 +673,192 @@ def KLIEP(
+    of equal conditional distributions p(x|y) for both source and target domains.
+
+    See Section 3 of [4]_ for details.
+
+    Parameters
+    ----------
+    gamma : float or array like
+        Parameters for the kernels.
+    random_state : int, RandomState instance or None, default=None
+        Determines random number generation for dataset creation. Pass an int
+        for reproducible output across multiple function calls.
+
+    Attributes
+    ----------
+    `X_source_` : array-like, shape (n_samples, n_features)
+        The source data used for fitting.
+    `y_source_` : array-like, shape (n_samples,)
+        The source labels used for fitting.
+
+    References
+    ----------
+    .. [4] Kun Zhang et. al. Domain Adaptation under Target and Conditional Shift
+           In ICML, 2013.
+    """
+
+    def __init__(self, gamma):
+        super().__init__()
+        self.gamma = gamma
+
+    def fit(self, X, y, sample_domain=None, **kwargs):
+        """Fit adaptation parameters.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The source data.
+        y : array-like, shape (n_samples,)
+            The source labels.
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels (same as sample_domain).
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, y, sample_domain = check_X_y_domain(X, y, sample_domain)
+        X_source, _, y_source, _ = source_target_split(
+            X, y, sample_domain=sample_domain
+        )
+        self.X_source_fit_, self.y_source_fit_ = X_source, y_source
+        return self
+
+    def adapt(self, X, y=None, sample_domain=None, **kwargs):
+        """Predict adaptation (weights, sample or labels).
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The source data.
+        y : array-like, shape (n_samples,)
+            The source labels.
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels (same as sample_domain).
+
+        Returns
+        -------
+        X_t : array-like, shape (n_samples, n_components)
+            The data (same as X).
+        y_t : array-like, shape (n_samples,)
+            The labels (same as y).
+        weights : array-like, shape (n_samples,)
+            The weights of the samples.
+        """
+        X, sample_domain = check_X_domain(X, sample_domain)
+        X_source, X_target = source_target_split(X, sample_domain=sample_domain)
+
+        m = X_source.shape[0]
+        n = X_target.shape[0]
+
+        # if no source data, return uniform weights for target
+        if m == 0:
+            return AdaptationOutput(X=X, y=y, sample_weights=np.zeros(n))
+
+        # if no y provided, get the source labels from fit method
+        if y is None:
+            np.testing.assert_array_equal(
```

Comment:
Following the PR #94, the optimization is done in the `.fit` and update if `X_source != self.X_source_fit_` is it possible to have the same ?